src/Type/Data/Symbol.purs

```diff
@@ -36,3 +41,35 @@ instance equalsSymbol
 equals :: forall l r o. Equals l r o => SProxy l -> SProxy r -> BProxy o
 equals _ _ = BProxy
 
+
+-- Mirror
+
+class MirrorP (a :: Symbol) (mirror :: Symbol) | a -> mirror, mirror -> a
```

Comment:
The `mirror -> a` functional dependency here doesn't seem to be respected by the `mirrorCons` instance.  Possibly makes sense to keep `MirrorP` unidirectional anyway, if `Mirror` runs it in both directions.

Going one step further: another option is to only have the one class, and keep it unidirectional.
Because `Mirror` is its own inverse, users can swap the args if they need it to run it the other direction.  If a users wants both directions, they can use two constraints like you did in `mirrorMirrorP`.

Reply:
Is has been some time since I wrote this code, so I need to retrace my steps here.
If I remember correctly the advantage of having Mirror be bidirectional is, that type inference works a lot better.